roles/openshift_health_checker/library/aos_version.py

```diff
@@ -94,17 +102,34 @@ def _retrieve_available_packages(expected_pkgs):
     # attempt to determine what packages are available via yum they may
     # be excluded. So, for our purposes here, disable excludes to see
     # what will really be available during an install or upgrade.
-    yb.conf.disable_excludes = ['all']
 
-    try:
-        pkgs = yb.pkgSack.returnPackages(patterns=expected_pkgs)
-    except yum.Errors.PackageSackError as excinfo:
-        # you only hit this if *none* of the packages are available
-        raise AosVersionException('\n'.join([
-            'Unable to find any OpenShift packages.',
-            'Check your subscription and repo settings.',
-            str(excinfo),
-        ]))
+    if PKG_MGR == "yum":
+        # search for package versions available for openshift pkgs
+        yb = yum.YumBase()  # pylint: disable=invalid-name
+
+        yb.conf.disable_excludes = ['all']
+
+        try:
+            pkgs = yb.pkgSack.returnPackages(patterns=expected_pkgs)
+        except yum.Errors.PackageSackError as excinfo:
+            # you only hit this if *none* of the packages are available
+            raise AosVersionException('\n'.join([
+                'Unable to find any OpenShift packages.',
+                'Check your subscription and repo settings.',
+                str(excinfo),
+            ]))
+    elif PKG_MGR == "dnf":
+        dbase = dnf.Base()  # pyling: disable=invalid-name
+
+        dbase.conf.disable_excludes = ['all']
+        dbase.read_all_repos()
+        dbase.fill_sack(load_system_repo=False, load_available_repos=True)
+
+        dquery = dbase.sack.query()
+        aquery = dquery.available()
+
+        pkgs = list(aquery.filter(name=expected_pkgs))
```

Comment:
I've looked at this a little more closely.

It seems the yum version is also broken, and that this plugin only works as intended if you're using versioned packages.  It seems an error is only raised for yum if there's a problem connection to the repo; it will happily return a list of zero packages.

A list of zero packages is not checked anywhere unless we're using versioned packages.

Additionally, checking the list for only for zero-length results in an error case as well.

For instance:
```
pkgs = yb.pkgSack.returnPackages(patterns=['yum', 'gcc99x'])
```
gcc99x is not a real package.  pkgs will return a list of yum packages; so the list is greater than zero length, but we didn't actually find all the packages we were looking for.

We're probably not hitting or seeing this bug currently because the packages are probably in the same repo together, or we might be using versions, IDK.  We should probably iterate through the package list, create a dictionary with available packages with key = desired_package, and return that to the calling function; calling function or this function can check if any of the values for the dictionary keys (package names) is zero length.  This will allow us to ensure that each desired package is available, and possibly each desired version of each package.

Maybe that's all oustide the scope of this PR.